src/test/java/uk/gov/pay/connector/tasks/HistoricalEventEmitterWorkerTest.java

```diff
@@ -138,4 +152,46 @@ public void executeShouldIgnoreEventIfStateTransitionIsNotFound() {
         verify(emittedEventDao, times(1)).recordEmission(daoArgumentCaptor.capture());
         assertThat(daoArgumentCaptor.getAllValues().get(0).getEventType(), is("PAYMENT_CREATED"));
     }
+
+    @Test
+    public void executeShouldEmitManualEventsWithTerminalAuthenticationState() throws QueueException {
+        ChargeEventEntity firstEvent = ChargeEventEntityFixture.aValidChargeEventEntity()
+                .withTimestamp(ZonedDateTime.now().plusMinutes(1))
+                .withCharge(chargeEntity)
+                .withChargeStatus(ChargeStatus.ENTERING_CARD_DETAILS)
+                .build();
+
+        ChargeEventEntity secondEvent = ChargeEventEntityFixture.aValidChargeEventEntity()
+                .withTimestamp(ZonedDateTime.now().plusMinutes(2))
+                .withCharge(chargeEntity)
+                .withChargeStatus(ChargeStatus.AUTHORISATION_SUCCESS)
+                .build();
+
+        chargeEntity.getEvents().add(firstEvent);
+        chargeEntity.getEvents().add(secondEvent);
+
+        when(chargeDao.findMaxId()).thenReturn(1L);
+        when(chargeDao.findById(1L)).thenReturn(Optional.of(chargeEntity));
+
+        worker.execute(1L, OptionalLong.empty());
+
+        verify(eventQueue).emitEvent(any(PaymentDetailsEntered.class));
+    }
+
+    @Test public void exectueShouldNotEmitManualEventsWithNoTerminalAuthenticationState() throws QueueException {
```

Comment:
`throws QueueException ` not required

```suggestion
    @Test public void executeShouldNotEmitManualEventsWithNoTerminalAuthenticationState() throws QueueException {
```